### ISSUE TITLE
Need to Consider the Case Senstive feature in CLI while creating new file.

### ISSUE THREAD

**Issue by AayushSaini101** [asyncapi/cli#790]
#### Reason/Context
Please try answering a few of those questions
- Why do we need this improvement?
 CLI should be case-sensitive while creating the file, currently, it is treating tutorial.yml and Tutorial.yml the same and enables to creation of files.
![image](https://github.com/asyncapi/cli/assets/60972989/904d8e9f-d2a5-4df4-be8e-c626ecbae22b)

- How will this change help?
It will always be optimal to consider the case-sensitive feature in the CLI

#### Description
Please try answering a few of those questions

- Will this be a breaking change?
- No, it will not break any current changes only some validation added.

**Comment by Souvikns** [comment]
I think this is where the issues is https://github.com/asyncapi/cli/blob/2cc269700fdca55f65bd5ed0d63a1c2726d25011/src/commands/new/file.ts#L145


https://stackoverflow.com/questions/27367261/check-if-file-exists-case-sensitive
https://forums.raspberrypi.com/viewtopic.php?t=119198

**Comment by AayushSaini101** [comment]
Thanks, @Souvikns Yes, here is the issue. that need to be changed,  but, If we write the logic to create a new file after comparing the case as well, we cannot because OS doesn't allow to creation of two separate files with different cases.

**Comment by Souvikns** [comment]
Yeah could we change the `file-name` that we get as input to lowercase and then create the file, or will it create inconvenience for the user?

**Comment by AayushSaini101** [comment]
@Souvikns We can see that when the user tries to create a file of the same name but different syntax it should get a message that a particular file is present with a different case. Please change the name. What's your thought?

**Comment by AayushSaini101** [comment]
> @Souvikns We can do this when the user tries to create a file of the same name but different case it should get a message that a particular file is present with a different case. Please change the name. What's your thought?

@Souvikns @fmvilas Any thoughts on this?

**Comment by Souvikns** [comment]
> @Souvikns We can do this when the user tries to create a file of the same name but different case it should get a message that a particular file is present with a different case. Please change the name. What's your thought?


🤔 actually the less warning from CLI the better the user experience, it would be better to handle this issue internally, and in our case, we can't really save two same file names with lower and uppercase because of how the filesystem works in nodejs. In my opinion, the error message we have right now works fine, and no need to change anything.

**Comment by derberg** [comment]
wasn't it solved?

**Comment by peter-rr** [comment]
> 🤔 actually the less warning from CLI the better the user experience, it would be better to handle this issue internally, and in our case, we can't really save two same file names with lower and uppercase because of how the filesystem works in nodejs. In my opinion, the error message we have right now works fine, and no need to change anything.

I agree with this approach 👍 

@Souvikns Shouldn't we convert the input to lowercase as you proposed [here](https://github.com/asyncapi/cli/issues/790#issuecomment-1715385152)?

**Comment by SagarSharma2809** [comment]
> Thanks, @Souvikns Yes, here is the issue. that need to be changed, but, If we write the logic to create a new file after comparing the case as well, we cannot because OS doesn't allow to creation of two separate files with different cases.

Hi, new to Async API sorry if there is anything wrong with my ques. Can i ask why the OS doesn't allow case sensitive files? Wouldn't it be better user experience?

**Comment by SagarSharma2809** [comment]
> > @Souvikns We can do this when the user tries to create a file of the same name but different case it should get a message that a particular file is present with a different case. Please change the name. What's your thought?
> 
> 🤔 actually the less warning from CLI the better the user experience, it would be better to handle this issue internally, and in our case, we can't really save two same file names with lower and uppercase because of how the filesystem works in nodejs. In my opinion, the error message we have right now works fine, and no need to change anything.

Agreed that the less warnings the better user experience but since many doesn't/ will not know the OS case-insensitivity they will get confused as to why different name is not working and if Tutorial.txt already exist. Thus, will try TUTORIAL.txt or something and fail. 

Maybe, we can give a better CLI response like " Error: A file named "tutorial.txt" already exists in the directory.

Your current file system is case-insensitive, so "Tutorial.txt" would be treated as the same file name.

Please choose a different name for your new file or delete/rename the existing file."

**Comment by SagarSharma2809** [comment]
> > > @Souvikns We can do this when the user tries to create a file of the same name but different case it should get a message that a particular file is present with a different case. Please change the name. What's your thought?
> > 
> > 
> > 🤔 actually the less warning from CLI the better the user experience, it would be better to handle this issue internally, and in our case, we can't really save two same file names with lower and uppercase because of how the filesystem works in nodejs. In my opinion, the error message we have right now works fine, and no need to change anything.
> 
> Agreed that the less warnings the better user experience but since many doesn't/ will not know the OS case-insensitivity they will get confused as to why different name is not working and if Tutorial.txt already exist. Thus, will try TUTORIAL.txt or something and fail.
> 
> Maybe, we can give a better CLI response like " Error: A file named "tutorial.txt" already exists in the directory.
> 
> Your current file system is case-insensitive, so "Tutorial.txt" would be treated as the same file name.
> 
> Please choose a different name for your new file or delete/rename the existing file."

I am willing to change the CLI response if this idea sounds ok

**Comment by IITI-tushar** [comment]
Hi @AayushSaini101 @derberg ,
 If this issue still relevant .Can I work on this?

**Comment by harshita9104** [comment]
Hello @AayushSaini101 , @derberg ,

I hope this message finds you well!
This issue interests me , could you please assign it to me!

Thank you!

**Comment by Adi-204** [comment]
@Souvikns @Shurtu-gal 
This issue is pending for long time. Can you please provide feedback on which idea we can implement? 
One approach can be - Save all files in lowercase and give warning 'particular file is present with a different case. Please change the name.' when user try to create a file with some other case ?

**Comment by AayushSaini101** [comment]
No plan to implement this feature in this CLI.
